### DISCUSSION TITLE
What does E409 mean and how do I resolve it?

### DISCUSSION THREAD

**Discussion by ibra** [openupm/openupm#2873] [Q&A] [answered]
Hey! I'm the maintainer of [Ibralogue](https://github.com.com/ibra/Ibralogue) which is a dialogue library for unity. For the past few releases of Ibralogue, i increment the version (manually) in the [package.json](https://github.com/Ibralogue/Ibralogue/blob/master/package.json) after a new release but despite me doing so this shows up on the [build pipelines page](https://openupm.com/packages/com.ibra.ibralogue/?subPage=pipelines).
![image](https://user-images.githubusercontent.com/61324615/160292018-b2487497-628f-4495-a840-85b555184601.png)

Any clues on how to resolve this?

**Comment by favoyang** [comment, marked as answer]
It means the Git tag you feed openupm contains a package.json with a version that already exists on the registry server.

In your case, tag v0.8.1's package.json is version 0.7.0. And of course 0.7.0 has been published months ago. https://github.com/Ibralogue/Ibralogue/blob/003ff21b2d17fed334a64e19ccea17b99941dfb2/package.json

To fix it, you can bump package.json to 0.8.2 first, then tag it as v0.8.2.

**Reply by lmtss** [reply]
Hello, I encountered a similar issue as well. I updated the version in package.json before tagging, but faced the same problem as shown in [this build log](https://dev.azure.com/openupm/openupm/_build/results?view=logs&buildId=50236&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=05a7d88e-decf-5c29-7849-5d16879a64a4). This new version (2.0.0-beta.9) has never been released before, so it shouldn't exist on the server.     
Although the package page at [this link](https://openupm.com/packages/com.motphys.rigidbody/) still shows the old version (beta.8), I am able to use Unity Package Manager to correctly reference the new version without any version-related issues.     
Or maybe I misunderstood? Is it possible that even without tagging, the package can still be uploaded to the server?

**Reply by favoyang** [reply]
> Is it possible that even without tagging, the package can still be uploaded to the server?

No.

![image](https://github.com/user-attachments/assets/5ea8788d-f008-401d-91ac-cb914555bcc4)

The version `2.0.0-beta.9` and `2.0.0-beta.7` are built with Git tag `com.motphys.rigidbody.native/2.0.0-beta.5` and `com.motphys.rigidbody.native/2.0.0-beta.4`. The problem is that the `com.motphys.rigidbody` gitTagPrefix can applied to multiple Git tags:

```
com.motphys.rigidbody.native
com.motphys.rigidbody
```

A common practice is each package has its own prefix.

To fix the issue
- added a splash to gitTagPrefix:
https://github.com/openupm/openupm/commit/8baf97622cdd9dc4789f6bbbd400d75df40008d1
- issued a rebuild to align `2.0.0-beta.9` and `2.0.0-beta.7` with their Git tags: https://openupm.com/packages/com.motphys.rigidbody/?subPage=pipelines

BTW, I noticed `com.motphys.rigidbody.native/` will applied to both `com.motphys.rigidbody.native` and `com.motphys.rigidbody.native.standard.development`, but not Git tags start with `com.motphys.rigidbody.native.standard.development`, so I leave you to decide whether a new git tag for `com.motphys.rigidbody.native.standard.development` is needed. It's not necessary but recommended. If you create the new tag, you should also update the gitTagPrefix to `com.motphys.rigidbody.native.standard.development/` to align with the change.

**Reply by lmtss** [reply]
Thank you for your detailed response; it was incredibly helpful!
